From 272148fbe56b0e5494e6a84597df7eba619825f0 Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Tue, 19 Sep 2023 09:09:37 +0200
Subject: [PATCH] qt6++

---
 PORT.QT6 | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/PORT.QT6 b/PORT.QT6
index f56fb4a571..4848777bc7 100644
--- a/PORT.QT6
+++ b/PORT.QT6
@@ -1,6 +1,12 @@
-GOAL: compile and run digiKam geolocation features with Qt6
+Context:
+--------
+
+Compile and run digiKam geolocation features with Qt6
 Port to Qt6 done by Gilles Caulier on September 2023
-Envirronement: Kubuntu 23.04 + Qt 6.4.2 + KF6 from master.
+Environment: Kubuntu 23.04 + Qt 6.4.2 + KF6 from master (https://invent.kde.org/graphics/digikam/-/tree/master/project/install-kf6?ref_type=heads).
+
+Port Details:
+-------------
 
 Only pure Qt codes are ported.
 Qt5 support is dropped.
@@ -10,6 +16,20 @@ KDE API based codes are disabled.
 Test codes not ported.
 Example codes not ported.
 
+Screenshots:
+------------
+
+Marble Qt6:
+    https://i.imgur.com/YJNp7H1.png
+    https://i.imgur.com/HVZBT3O.png
+
+digiKam Qt6 with Marble:
+    https://i.imgur.com/mBM7bqq.png
+    https://i.imgur.com/Y553qAG.png
+
+Directories Status:
+-------------------
+
 DIRECTORIES                                         STATUS
 .
 ├── cmake                                           -
-- 
GitLab

