From 2ef4ad1018bddbdd7e00517a36f5ed2e1314c27a Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Mon, 18 Sep 2023 18:32:31 +0200
Subject: [PATCH] qt6++

---
 PORT.QT6 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PORT.QT6 b/PORT.QT6
index d40986c942..99ddace26c 100644
--- a/PORT.QT6
+++ b/PORT.QT6
@@ -8,6 +8,7 @@ Only shared libraries are ported, not target applications.
 KDE API based codes are disabled.
 Test codes not ported.
 Example codes not ported.
+PhononQt6 feature not compiled.
 
 DIRECTORIES                                         STATUS
 .
-- 
GitLab

