diff -up marble-24.02.2/PORT.QT6.39~ marble-24.02.2/PORT.QT6
--- marble-24.02.2/PORT.QT6.39~	2024-04-16 17:22:21.514106296 +0200
+++ marble-24.02.2/PORT.QT6	2024-04-16 17:22:21.518106336 +0200
@@ -133,7 +133,7 @@ DIRECTORIES
 │   │   │   ├── shp                                 DISABLED: KDE BASED
 │   │   │   ├── traveling-salesman                  - Originally commented in CMakeLists.txt
 │   │   │   └── yours                               DONE
-│   │   └── templates                               DISABLED: NOT USED IN DIGIKAM
-│   └── thumbnailer                                 -
+│   │   └── templates                               - Originally commented in CMakeLists.txt
+│   └── thumbnailer                                 DISABLED: KDE BASED
 ├── tests                                           -
 └── tools                                           -
diff -up marble-24.02.2/src/plugins/CMakeLists.txt.39~ marble-24.02.2/src/plugins/CMakeLists.txt
--- marble-24.02.2/src/plugins/CMakeLists.txt.39~	2024-04-16 17:22:21.518106336 +0200
+++ marble-24.02.2/src/plugins/CMakeLists.txt	2024-04-16 17:22:43.026325877 +0200
@@ -11,7 +11,7 @@ add_definitions("-DLOGGING_IDENTIFIER=MA
 
 if(NOT ANDROID)
 #    add_subdirectory( designer )
-#    add_subdirectory( templates )
+    add_subdirectory( templates )
 endif()
 
 add_subdirectory( render )
