From 72de2c920e3de14e18426cb32ed4053825813478 Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Tue, 19 Sep 2023 11:39:27 +0200
Subject: [PATCH] qt6++

---
 MarbleMacros.cmake | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/MarbleMacros.cmake b/MarbleMacros.cmake
index 235ca0efcb..7c5fe96c4d 100644
--- a/MarbleMacros.cmake
+++ b/MarbleMacros.cmake
@@ -23,15 +23,15 @@ else()
 endif()
 
 macro(qt_add_resources)
-  qt5_add_resources(${ARGN})
+  qt6_add_resources(${ARGN})
 endmacro()
 
 macro(qt_wrap_ui)
-  qt5_wrap_ui(${ARGN})
+  qt6_wrap_ui(${ARGN})
 endmacro()
 
 macro(qt_generate_moc)
-  qt5_generate_moc(${ARGN})
+  qt6_generate_moc(${ARGN})
 endmacro()
 
 # the place to put in common cmake macros
@@ -182,8 +182,8 @@ else()
 if(TARGET Qt6::lconvert)
     set(lconvert_executable Qt6::lconvert)
 else()
-    # Qt < 5.3.1 does not define Qt6::lconvert
-    get_target_property(lrelease_location Qt6::lrelease LOCATION)
+    # Qt < 5.3.1 does not define Qt5::lconvert
+    get_target_property(lrelease_location Qt5::lrelease LOCATION)
     get_filename_component(lrelease_path ${lrelease_location} PATH)
     find_program(lconvert_executable
         NAMES lconvert-qt5 lconvert
-- 
GitLab

