From a114e3e915082356b60c4fd19f35e48837efb345 Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Mon, 18 Sep 2023 09:51:19 +0200
Subject: [PATCH] Qt6++ : libmarblewidget-qt5 compile and link

---
 src/lib/marble/TinyWebBrowser.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/marble/TinyWebBrowser.cpp b/src/lib/marble/TinyWebBrowser.cpp
index 97cd2bc4b1..642834ab4b 100644
--- a/src/lib/marble/TinyWebBrowser.cpp
+++ b/src/lib/marble/TinyWebBrowser.cpp
@@ -76,7 +76,7 @@ void TinyWebBrowser::print()
 
     QPointer<QPrintDialog> dlg = new QPrintDialog( &printer, this );
     if ( dlg->exec() )
-        page()->print( &printer,  [=](bool){} );
+        QWebEngineView::print( &printer );
     delete dlg;
 #endif
 }
-- 
GitLab

