From a78fafb5d1796729de17f2e7963dce8cfe700269 Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Tue, 28 Nov 2023 07:55:19 +0100
Subject: [PATCH] fix broken qt6 signal/slot

---
 src/lib/marble/HttpJob.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/marble/HttpJob.cpp b/src/lib/marble/HttpJob.cpp
index 2c51eca303..2ca4ed6f0f 100644
--- a/src/lib/marble/HttpJob.cpp
+++ b/src/lib/marble/HttpJob.cpp
@@ -135,7 +135,7 @@ void HttpJob::execute()
 
     connect( d->m_networkReply, SIGNAL(downloadProgress(qint64,qint64)),
              SLOT(downloadProgress(qint64,qint64)));
-    connect( d->m_networkReply, SIGNAL(error(QNetworkReply::NetworkError)),
+    connect( d->m_networkReply, SIGNAL(errorOccurred(QNetworkReply::NetworkError)),
              SLOT(error(QNetworkReply::NetworkError)));
     connect( d->m_networkReply, SIGNAL(finished()),
              SLOT(finished()));
-- 
GitLab

