From b5424e7dacd2a59e448adcda42a2eb31eea1c9f0 Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Thu, 21 Sep 2023 11:01:39 +0200
Subject: [PATCH] polish

---
 PORT.QT6 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/PORT.QT6 b/PORT.QT6
index 9421b2e14a..dd91c19c04 100644
--- a/PORT.QT6
+++ b/PORT.QT6
@@ -2,8 +2,8 @@ Context:
 --------
 
 Compile and run digiKam geolocation features with Qt6
-Port to Qt6 done by Gilles Caulier on September 2023
-Environment: Kubuntu 23.04 + Qt 6.4.2 + KF6 from master (https://invent.kde.org/graphics/digikam/-/tree/master/project/install-kf6?ref_type=heads).
+Port to Qt6 done by cgilles on September 2023 (https://invent.kde.org/cgilles)
+Environment: Kubuntu 23.04 + Qt 6.5.3 + KF6 from master (https://invent.kde.org/graphics/digikam/-/tree/master/project/install-kf6?ref_type=heads).
 
 Port Details:
 -------------
-- 
GitLab

