diff -up marble-24.02.2/PORT.QT6.60~ marble-24.02.2/PORT.QT6
--- marble-24.02.2/PORT.QT6.60~	2024-04-16 17:26:18.620532086 +0200
+++ marble-24.02.2/PORT.QT6	2024-04-16 17:26:18.628532181 +0200
@@ -76,8 +76,8 @@ DIRECTORIES
 │   │       │   └── instructions                    DONE
 │   │       └── webpopup                            -
 │   ├── mac                                         -
-│   ├── plasma                                      -
-│   ├── plasmarunner                                -
+│   ├── plasma                                      DISABLED: KDE based codes.
+│   ├── plasmarunner                                DISABLED: KDE based codes.
 │   ├── plugins
 │   │   ├── designer                                DONE
 │   │   │   ├── latloneditwidget                    DONE
diff -up marble-24.02.2/src/CMakeLists.txt.60~ marble-24.02.2/src/CMakeLists.txt
--- marble-24.02.2/src/CMakeLists.txt.60~	2024-04-16 17:26:18.628532181 +0200
+++ marble-24.02.2/src/CMakeLists.txt	2024-04-16 17:26:38.196764382 +0200
@@ -76,8 +76,8 @@ if(APPLE)
   add_subdirectory( mac )
 endif(APPLE)
 
-add_subdirectory( plasma )
-add_subdirectory( plasmarunner )
+#add_subdirectory( plasma )
+#add_subdirectory( plasmarunner )
 add_subdirectory( thumbnailer )
 
 if(ECM_FOUND)
