From f9e5ca1a6489b81bbca8751ba5110c0ebfbd2b2f Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Mon, 18 Sep 2023 08:12:29 +0200
Subject: [PATCH] qt6++

---
 PORT.QT6 | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/PORT.QT6 b/PORT.QT6
index ab01c8af14..ef1d5865b5 100644
--- a/PORT.QT6
+++ b/PORT.QT6
@@ -75,26 +75,26 @@ DIRECTORIES                                         STATUS
 │   ├── lib
 │   │   ├── astro
 │   │   └── marble
-│   │       ├── blendings
+│   │       ├── blendings                           DONE
 │   │       ├── cloudsync
 │   │       ├── declarative
 │   │       ├── geodata
 │   │       │   ├── data                            DONE
 │   │       │   │   └── tests
-│   │       │   ├── graphicsitem
-│   │       │   ├── handlers
+│   │       │   ├── graphicsitem                    DONE
+│   │       │   ├── handlers                        DONE
 │   │       │   │   ├── dgml                        DONE
 │   │       │   │   └── kml                         DONE
-│   │       │   ├── parser
-│   │       │   ├── scene
-│   │       │   ├── writer
+│   │       │   ├── parser                          DONE
+│   │       │   ├── scene                           DONE
+│   │       │   ├── writer                          DONE
 │   │       │   └── writers
 │   │       │       ├── dgml                        DONE
 │   │       │       └── kml                         DONE
-│   │       ├── graphicsview
+│   │       ├── graphicsview                        DONE
 │   │       ├── htmlfeatures
 │   │       ├── layers                              DONE
-│   │       ├── osm
+│   │       ├── osm                                 DONE
 │   │       ├── projections
 │   │       ├── routing
 │   │       │   └── instructions
-- 
GitLab

