diff -up marble-24.05.0/CMakeLists.txt.6~ marble-24.05.0/CMakeLists.txt
--- marble-24.05.0/CMakeLists.txt.6~	2024-05-28 20:28:59.201515437 +0200
+++ marble-24.05.0/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -110,7 +110,7 @@ set_package_properties(Protobuf PROPERTI
 if(ANDROID)
     set(REQUIRED_QT_VERSION 5.14.0)
    # TODO: still needed with ECM toolchain?
-   SET(QT_QMAKE_EXECUTABLE "$ENV{Qt5_android}/bin/qmake")
+   SET(QT_QMAKE_EXECUTABLE "$ENV{Qt6_android}/bin/qmake")
 else()
     set(REQUIRED_QT_VERSION 6.4.2)
 endif()
@@ -131,8 +131,8 @@ find_package(Qt6 ${REQUIRED_QT_VERSION}
 )
 
 if(ANDROID)
-    find_package(Qt5Positioning ${REQUIRED_QT_VERSION} REQUIRED)
-    find_package(Qt5Multimedia ${REQUIRED_QT_VERSION} REQUIRED)
+    find_package(Qt6Positioning ${REQUIRED_QT_VERSION} REQUIRED)
+    find_package(Qt6Multimedia ${REQUIRED_QT_VERSION} REQUIRED)
 
    set ( MARBLE_NO_WEBKITWIDGETS TRUE )
 else()
@@ -314,7 +314,7 @@ set(KDE_INSTALL_DIRS_NO_DEPRECATED TRUE)
 set(KDE_SKIP_UNINSTALL_TARGET ON CACHE BOOL "KDE uninstall target must be disabled")
 set(REQUIRED_KF6_MIN_VERSION 5.240.0)
 
-find_package(ECM ${REQUIRED_ECM_VERSION} QUIET)
+find_package(ECM ${REQUIRED_ECM_VERSION})
 if(ECM_FOUND)
   set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
 
@@ -385,7 +385,7 @@ marble_install_po_files_as_qm(poqm)
 # As a temporary hack (your task to improve it, dear reader :) ) the macros
 # are called only from src/apps/marble-kde/CMakeLists.txt.
 # That should catch 99.9 % cases of builds from released tarball, where all the
-# KF5-dependent apps and plugins will be built and installed together.
+# KF6-dependent apps and plugins will be built and installed together.
 # Currently these are:
 # src/apps/marble-kde
 # src/plasmarunner
diff -up marble-24.05.0/MarbleMacros.cmake.6~ marble-24.05.0/MarbleMacros.cmake
--- marble-24.05.0/MarbleMacros.cmake.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/MarbleMacros.cmake	2024-05-28 20:28:59.229515740 +0200
@@ -82,7 +82,7 @@ macro( marble_add_test TEST_NAME )
         add_executable( ${TEST_NAME} ${${TEST_NAME}_SRCS} )
         target_link_libraries(${TEST_NAME}
             marblewidget
-            Qt5::Test
+            Qt6::Test
         )
 
         set_target_properties( ${TEST_NAME} PROPERTIES
@@ -153,9 +153,9 @@ endif()
 
 
 # Find Qt translation tools
-find_package(Qt5LinguistTools CONFIG)
+#find_package(Qt6LinguistTools CONFIG)
 
-if(NOT Qt5LinguistTools_FOUND)
+if(NOT Qt6LinguistTools_FOUND)
 
 # dummy implementation
 function(marble_install_po_files_as_qm podir)
@@ -179,11 +179,11 @@ endfunction()
 
 else()
 
-if(TARGET Qt5::lconvert)
-    set(lconvert_executable Qt5::lconvert)
+if(TARGET Qt6::lconvert)
+    set(lconvert_executable Qt6::lconvert)
 else()
-    # Qt < 5.3.1 does not define Qt5::lconvert
-    get_target_property(lrelease_location Qt5::lrelease LOCATION)
+    # Qt < 5.3.1 does not define Qt6::lconvert
+    get_target_property(lrelease_location Qt6::lrelease LOCATION)
     get_filename_component(lrelease_path ${lrelease_location} PATH)
     find_program(lconvert_executable
         NAMES lconvert-qt5 lconvert
@@ -219,7 +219,7 @@ function(marble_process_po_files_as_qm l
     add_custom_command(OUTPUT ${qm_file}
         COMMAND ${lconvert_executable}
             ARGS -i ${po_file} -o ${ts_file} -target-language ${lang}
-        COMMAND Qt5::lrelease
+        COMMAND Qt6::lrelease
             ARGS -removeidentical -nounfinished -silent ${ts_file} -qm ${qm_file}
         DEPENDS ${po_file}
     )
diff -up marble-24.05.0/bootstrap-marble.sh.6~ marble-24.05.0/bootstrap-marble.sh
--- marble-24.05.0/bootstrap-marble.sh.6~	2024-05-28 20:28:59.221515654 +0200
+++ marble-24.05.0/bootstrap-marble.sh	2024-05-28 20:28:59.229515740 +0200
@@ -17,20 +17,16 @@ fi
 
 cd build
 
-export MarbleOptions=" \
--DWITH_DESIGNER_PLUGIN=OFF \
--DBUILD_MARBLE_TESTS=OFF \
--DBUILD_MARBLE_TOOLS=OFF \
--DBUILD_MARBLE_EXAMPLES=OFF \
--DBUILD_MARBLE_APPS=OFF \
--DBUILD_WITH_DBUS=OFF \
--DQTONLY=ON \
--DBUILD_TESTING=OFF \
-"
-
 cmake -G "$MAKEFILES_TYPE" \
       -DCMAKE_INSTALL_PREFIX=/usr \
+      -DWITH_DESIGNER_PLUGIN=OFF \
+      -DBUILD_MARBLE_TESTS=OFF \
+      -DBUILD_MARBLE_TOOLS=OFF \
+      -DBUILD_MARBLE_EXAMPLES=OFF \
+      -DBUILD_MARBLE_APPS=OFF \
+      -DBUILD_WITH_DBUS=OFF \
+      -DQTONLY=ON \
+      -DBUILD_TESTING=OFF \
       -Wno-dev \
-      $MarbleOptions \
       ..
 
diff -up marble-24.05.0/doc/CMakeLists.txt.6~ marble-24.05.0/doc/CMakeLists.txt
--- marble-24.05.0/doc/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/doc/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,8 +9,8 @@ endif()
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS DocTools)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS DocTools)
 
-if(KF5_FOUND)
+if(KF6_FOUND)
     kdoctools_create_handbook(index.docbook INSTALL_DESTINATION ${KDE_INSTALL_DOCBUNDLEDIR}/en SUBDIR marble)
 endif()
diff -up marble-24.05.0/examples/cpp/animation-video/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/animation-video/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/animation-video/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/animation-video/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,19 +2,19 @@ CMAKE_MINIMUM_REQUIRED (VERSION 2.6)
 SET (TARGET animation-video)
 PROJECT (${TARGET})
 
-FIND_PACKAGE (Qt5Core REQUIRED)
-FIND_PACKAGE (Qt5Widgets REQUIRED)
+FIND_PACKAGE (Qt6Core REQUIRED)
+FIND_PACKAGE (Qt6Widgets REQUIRED)
 FIND_PACKAGE (Marble REQUIRED)
 FIND_PACKAGE (OpenCV REQUIRED)
 
 INCLUDE_DIRECTORIES (
-${Qt5Core_INCLUDE_DIRS}
-${Qt5Widgets_INCLUDE_DIRS}
+${Qt6Core_INCLUDE_DIRS}
+${Qt6Widgets_INCLUDE_DIRS}
 ${MARBLE_INCLUDE_DIR}
 ${OpenCV_INCLUDE_DIRS}
 )
 
-SET (LIBS ${MARBLE_LIBRARIES} ${Qt5Core_LIBRARIES} ${Qt5Widgets_LIBRARIES} ${OpenCV_LIBRARIES})
+SET (LIBS ${MARBLE_LIBRARIES} ${Qt6Core_LIBRARIES} ${Qt6Widgets_LIBRARIES} ${OpenCV_LIBRARIES})
 
 ADD_EXECUTABLE (${TARGET} main.cpp)
 TARGET_LINK_LIBRARIES (${TARGET} ${LIBS})
diff -up marble-24.05.0/examples/cpp/basic-routing/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/basic-routing/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/basic-routing/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/basic-routing/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(basic-routing)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(basic-routing  main.cpp)
-target_link_libraries(basic-routing  Marble Qt5::Widgets)
+target_link_libraries(basic-routing  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/drawing-places/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/drawing-places/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/drawing-places/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/drawing-places/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(drawing-places)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(drawing-places main.cpp)
-target_link_libraries(drawing-places  Marble Qt5::Widgets)
+target_link_libraries(drawing-places  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/geopainter/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/geopainter/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/geopainter/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/geopainter/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(geopainter)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(geopainter  main.cpp)
-target_link_libraries(geopainter  Marble Qt5::Widgets)
+target_link_libraries(geopainter  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/hello-marble/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/hello-marble/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/hello-marble/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/hello-marble/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(hello-marble)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(hello-marble  main.cpp)
-target_link_libraries(hello-marble  Marble Qt5::Widgets)
+target_link_libraries(hello-marble  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/kml-inspector/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/kml-inspector/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/kml-inspector/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/kml-inspector/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(kml-inspector)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(kml-inspector  main.cpp)
-target_link_libraries(kml-inspector  Marble Qt5::Widgets)
+target_link_libraries(kml-inspector  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/map-controls/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/map-controls/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/map-controls/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/map-controls/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(map-controls)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(map-controls  main.cpp)
-target_link_libraries(map-controls  Marble Qt5::Widgets)
+target_link_libraries(map-controls  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/map-properties/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/map-properties/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/map-properties/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/map-properties/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(map-properties)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(map-properties  main.cpp)
-target_link_libraries(map-properties  Marble Qt5::Widgets)
+target_link_libraries(map-properties  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt.6~ marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt
--- marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -33,7 +33,7 @@ target_link_libraries(
     ${TARGET}
     marblewidget
     marbleui
-    Qt5::Quick
+    Qt6::Quick
 )
 
 install( TARGETS ${TARGET} RUNTIME DESTINATION ${MARBLE_EXAMPLES_INSTALL_DIR}/${TARGET} )
diff -up marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/marble-game/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(marble-game)
 
-find_package(Qt5Widgets REQUIRED)
-find_package(Qt5Quick REQUIRED)
+find_package(Qt6Widgets REQUIRED)
+find_package(Qt6Quick REQUIRED)
 find_package(Marble REQUIRED)
 
 set(CMAKE_AUTOMOC TRUE)
@@ -25,6 +25,6 @@ add_executable(marble-game ${marble-game
 
 target_link_libraries(marble-game
     Marble
-    Qt5::Widgets
-    Qt5::Quick
+    Qt6::Widgets
+    Qt6::Quick
 )
diff -up marble-24.05.0/examples/cpp/marbleQuick2/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/marbleQuick2/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/marbleQuick2/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/marbleQuick2/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,9 +2,9 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(marbleQuick)
 
-find_package(Qt5Widgets REQUIRED)
-find_package(Qt5Quick REQUIRED)
+find_package(Qt6Widgets REQUIRED)
+find_package(Qt6Quick REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(marbleQuick  main.cpp)
-target_link_libraries(marbleQuick  Marble Qt5::Widgets Qt5::Quick)
+target_link_libraries(marbleQuick  Marble Qt6::Widgets Qt6::Quick)
diff -up marble-24.05.0/examples/cpp/overlay-rendering/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/overlay-rendering/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/overlay-rendering/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/overlay-rendering/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(overlay-rendering)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(overlay-rendering  main.cpp)
-target_link_libraries(overlay-rendering  Marble Qt5::Widgets)
+target_link_libraries(overlay-rendering  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/reverse-geocoding/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/reverse-geocoding/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/reverse-geocoding/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/reverse-geocoding/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(reverse-geocoding)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(reverse-geocoding  main.cpp)
-target_link_libraries(reverse-geocoding  Marble Qt5::Widgets)
+target_link_libraries(reverse-geocoding  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/search/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/search/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/search/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/search/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(search)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 add_executable(search  main.cpp)
-target_link_libraries(search  Marble Qt5::Widgets)
+target_link_libraries(search  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/squad-interpolation/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/squad-interpolation/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/squad-interpolation/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/squad-interpolation/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,10 +2,10 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(squad-interpolation)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 set(CMAKE_AUTOMOC TRUE)
 
 add_executable(squad-interpolation  squad-interpolation.cpp)
-target_link_libraries(squad-interpolation  Marble Qt5::Widgets)
+target_link_libraries(squad-interpolation  Marble Qt6::Widgets)
diff -up marble-24.05.0/examples/cpp/tour-preview/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/tour-preview/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/tour-preview/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/tour-preview/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,19 +2,19 @@ CMAKE_MINIMUM_REQUIRED (VERSION 2.6)
 SET (TARGET tour-preview)
 PROJECT (${TARGET})
 
-FIND_PACKAGE (Qt5Core REQUIRED)
-FIND_PACKAGE (Qt5Widgets REQUIRED)
+FIND_PACKAGE (Qt6Core REQUIRED)
+FIND_PACKAGE (Qt6Widgets REQUIRED)
 FIND_PACKAGE (Marble REQUIRED)
 FIND_PACKAGE (OpenCV REQUIRED)
 
 INCLUDE_DIRECTORIES (
-${Qt5Core_INCLUDE_DIRS}
-${Qt5Widgets_INCLUDE_DIRS}
+${Qt6Core_INCLUDE_DIRS}
+${Qt6Widgets_INCLUDE_DIRS}
 ${MARBLE_INCLUDE_DIR}
 ${OpenCV_INCLUDE_DIRS}
 )
 
-SET (LIBS ${MARBLE_LIBRARIES} ${Qt5Core_LIBRARIES} ${Qt5Widgets_LIBRARIES} ${OpenCV_LIBRARIES})
+SET (LIBS ${MARBLE_LIBRARIES} ${Qt6Core_LIBRARIES} ${Qt6Widgets_LIBRARIES} ${OpenCV_LIBRARIES})
 
 ADD_EXECUTABLE (${TARGET} tour-preview.cpp)
 TARGET_LINK_LIBRARIES (${TARGET} ${LIBS})
diff -up marble-24.05.0/examples/cpp/vehicletracking/CMakeLists.txt.external.6~ marble-24.05.0/examples/cpp/vehicletracking/CMakeLists.txt.external
--- marble-24.05.0/examples/cpp/vehicletracking/CMakeLists.txt.external.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/examples/cpp/vehicletracking/CMakeLists.txt.external	2024-05-28 20:28:59.229515740 +0200
@@ -2,10 +2,10 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(vehicletracking)
 
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6Widgets REQUIRED)
 find_package(Marble REQUIRED)
 
 set(CMAKE_AUTOMOC TRUE)
 
 add_executable(vehicletracking  vehicletracking.cpp)
-target_link_libraries(vehicletracking  Marble Qt5::Widgets)
+target_link_libraries(vehicletracking  Marble Qt6::Widgets)
diff -up marble-24.05.0/install/INSTALL.6~ marble-24.05.0/install/INSTALL
--- marble-24.05.0/install/INSTALL.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/install/INSTALL	2024-05-28 20:28:59.229515740 +0200
@@ -16,7 +16,7 @@ NOTE: This file describes how to build M
 Prerequisites for building Marble are:
  - Qt 5.3 or newer
  - Cmake version 2.8.12 or newer
- - (optional): KDE Frameworks 5.7 or newer (KF5) from the KDE repositories.
+ - (optional): KDE Frameworks 5.7 or newer (KF6) from the KDE repositories.
  - other dependencies are optional
 
 A checkout of Marble. You can check out Marble using GIT via the command:
@@ -27,7 +27,7 @@ git clone git://anongit.kde.org/marble ~
 You can build and install Marble in two different ways:
 
 1. As a Qt-only application.
-2. As a KF5-based application, using KDE Frameworks 5 classes.
+2. As a KF6-based application, using KDE Frameworks 5 classes.
 
 Choose one of them and read below for how to do it.
 
@@ -47,19 +47,19 @@ directory called 'marble-build'.
 
 Go into the build directory and type
 
-  cmake -DWITH_KF5=FALSE ../marble
+  cmake -DWITH_KF6=FALSE ../marble
 
 By default, marble is installed into /usr/local.  If you want to
 install it into any other directory, add
 -DCMAKE_INSTALL_PREFIX=/path/to/install/into like in this example:
 
-  cmake -DWITH_KF5=FALSE -DCMAKE_INSTALL_PREFIX=~/apps ../marble
+  cmake -DWITH_KF6=FALSE -DCMAKE_INSTALL_PREFIX=~/apps ../marble
 
 If you have no write permission to the installation directory of cmake
 (because you have a system installed cmake) you need to add another
 parameter -DPACKAGE_ROOT_PREFIX=~/apps to the cmake command:
 
-  cmake -DWITH_KF5=FALSE -DCMAKE_INSTALL_PREFIX=~/apps \
+  cmake -DWITH_KF6=FALSE -DCMAKE_INSTALL_PREFIX=~/apps \
         -DPACKAGE_ROOT_PREFIX=~/apps ../marble
 
 To modify the position of the data directory at compile time, see the notes
@@ -84,7 +84,7 @@ and start Marble either from the menu or
   marble-qt
 
 
-2. Build Marble as a KF5-based application.
+2. Build Marble as a KF6-based application.
 
 2.1 Create a build directory
 
@@ -126,7 +126,7 @@ Use the cmake option -DMARBLE_DATA_PATH
 built-in system path that Marble gathers its data from:
 
 Example:
-  cmake -DWITH_KF5=FALSE -DMARBLE_DATA_PATH /mnt1/marble/data ~/marble
+  cmake -DWITH_KF6=FALSE -DMARBLE_DATA_PATH /mnt1/marble/data ~/marble
 
 3.2 "At Runtime"
 
diff -up marble-24.05.0/src/CMakeLists.txt.6~ marble-24.05.0/src/CMakeLists.txt
--- marble-24.05.0/src/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -10,7 +10,7 @@ endif()
 # disabled for now:
 # no public Marble QML plugins are installed, libmarbledeclarative is only a shared lib
 if (FALSE)
-if( Qt5Quick_FOUND )
+if( Qt6Quick_FOUND )
   OPTION(IGNORE_CMAKE_INSTALL_PREFIX_FOR_DECLARATIVE_PLUGINS "Install Marble Qt Declarative plugins to QT_IMPORTS_DIR even if that is out of the CMAKE_INSTALL_PREFIX")
   IF(QT_IMPORTS_DIR)
     SET(MARBLE_QT_IMPORTS_DIR ${QT_IMPORTS_DIR})
diff -up marble-24.05.0/src/apps/CMakeLists.txt.6~ marble-24.05.0/src/apps/CMakeLists.txt
--- marble-24.05.0/src/apps/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -7,7 +7,7 @@ if(BUILD_MARBLE_APPS)
     endif()
 
     add_subdirectory(behaim)
-    if (NOT Qt5_VERSION VERSION_LESS 5.7.0)
+    if (NOT Qt6_VERSION VERSION_LESS 5.7.0)
         add_subdirectory(marble-maps)
     endif()
 endif()
diff -up marble-24.05.0/src/apps/behaim/CMakeLists.txt.6~ marble-24.05.0/src/apps/behaim/CMakeLists.txt
--- marble-24.05.0/src/apps/behaim/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/behaim/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -6,7 +6,7 @@ include_directories(${CMAKE_CURRENT_BINA
 if (ANDROID)
     set(TARGET MarbleBehaim)
 
-    find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS AndroidExtras)
+    find_package(Qt6 ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS AndroidExtras)
 
     add_library (${TARGET} SHARED ${marble_SRCS} ${marble_QRCS})
 
@@ -15,20 +15,20 @@ if (ANDROID)
         ${TARGET}
         marblewidget
         astro
-        Qt5::Concurrent
-        Qt5::Xml
-        Qt5::Widgets
-        Qt5::PrintSupport
-        Qt5::Network
-        Qt5::Sql
-        Qt5::Svg
-        Qt5::Positioning
-        Qt5::AndroidExtras
-        Qt5::Multimedia
+        Qt6::Concurrent
+        Qt6::Xml
+        Qt6::Widgets
+        Qt6::PrintSupport
+        Qt6::Network
+        Qt6::Sql
+        Qt6::Svg
+        Qt6::Positioning
+        Qt6::AndroidExtras
+        Qt6::Multimedia
     )
 
-    if (TARGET Qt5::WebEngineWidgets)
-        target_link_libraries(${TARGET} Qt5::WebEngineWidgets Qt5::WebEngine)
+    if (TARGET Qt6::WebEngineWidgets)
+        target_link_libraries(${TARGET} Qt6::WebEngineWidgets Qt6::WebEngine)
     endif()
 else()
     set(TARGET marble-behaim)
@@ -50,7 +50,7 @@ if (ANDROID)
         set(ABSOLUTE_INSTALL_PATH "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_PREFIX}")
     endif()
     get_filename_component(ABSOLUTE_INSTALL_PATH "${ABSOLUTE_INSTALL_PATH}" ABSOLUTE)
-    get_filename_component(QT_ANDROID_QT_ROOT "${Qt5Core_DIR}/../../.." ABSOLUTE)
+    get_filename_component(QT_ANDROID_QT_ROOT "${Qt6Core_DIR}/../../.." ABSOLUTE)
     set(ANDROID_SDK_ROOT $ENV{ANDROID_SDK_ROOT})
     set(ANDROID_NDK_ROOT $ENV{ANDROID_NDK_ROOT})
     set(QT_ANDROID_APP_PATH "${ABSOLUTE_INSTALL_PATH}/libs/${ANDROID_ABI}/libMarbleBehaim.so")
diff -up marble-24.05.0/src/apps/behaim/create-apk.py.6~ marble-24.05.0/src/apps/behaim/create-apk.py
--- marble-24.05.0/src/apps/behaim/create-apk.py.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/behaim/create-apk.py	2024-05-28 20:28:59.229515740 +0200
@@ -65,9 +65,9 @@ class Filter(object):
 
         return []
 
-qtDir = os.environ.get('Qt5_android')
+qtDir = os.environ.get('Qt6_android')
 if qtDir is None:
-    print ('Please setup the Qt5_android environment variable point to your Qt installation')
+    print ('Please setup the Qt6_android environment variable point to your Qt installation')
     sys.exit(1)
 
 parser = argparse.ArgumentParser(description='Create an Android application package (APK) for Marble Maps')
diff -up marble-24.05.0/src/apps/marble-kde/CMakeLists.txt.6~ marble-24.05.0/src/apps/marble-kde/CMakeLists.txt
--- marble-24.05.0/src/apps/marble-kde/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-kde/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -2,9 +2,9 @@ if(NOT ECM_FOUND)
     return()
 endif()
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS Crash Parts CoreAddons I18n)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS Crash Parts CoreAddons I18n)
 
-if(NOT KF5_FOUND)
+if(NOT KF6_FOUND)
     return()
 endif()
 
@@ -22,9 +22,9 @@ set_target_properties(marble_part PROPER
 target_link_libraries(marble_part
     marbleui
     marblewidget
-    KF5::Parts
-    KF5::CoreAddons
-    KF5::I18n
+    KF6::Parts
+    KF6::CoreAddons
+    KF6::I18n
 )
 
 install(TARGETS marble_part DESTINATION ${KDE_INSTALL_PLUGINDIR})
@@ -49,10 +49,10 @@ target_link_libraries (
     marbleui
     marblewidget
     astro
-    KF5::Parts
-    KF5::Crash
-    KF5::CoreAddons
-    KF5::I18n
+    KF6::Parts
+    KF6::Crash
+    KF6::CoreAddons
+    KF6::I18n
     )
 
 ## this works for win32 only because FindQt does not handle this correct
@@ -74,8 +74,8 @@ install(FILES marble_worldwind.desktop D
 # See toplevel CMakeLists.txt for further explanation why it is done here
 ki18n_install(${CMAKE_SOURCE_DIR}/po)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS DocTools)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS DocTools)
 
-if (KF5DocTools_FOUND)
+if (KF6DocTools_FOUND)
     kdoctools_install(${CMAKE_SOURCE_DIR}/po)
 endif()
diff -up marble-24.05.0/src/apps/marble-kde/kdemain.cpp.6~ marble-24.05.0/src/apps/marble-kde/kdemain.cpp
--- marble-24.05.0/src/apps/marble-kde/kdemain.cpp.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-kde/kdemain.cpp	2024-05-28 20:28:59.229515740 +0200
@@ -83,7 +83,7 @@ int main ( int argc, char *argv[] )
 
     // Load Qt translation system catalog for the plugins and libmarblewidget
     loadLibAndPluginTranslations(app);
-    // Init KF5 translation system
+    // Init KF6 translation system
     KLocalizedString::setApplicationDomain("marble");
 
 
diff -up marble-24.05.0/src/apps/marble-kde/marble_part.cpp.6~ marble-24.05.0/src/apps/marble-kde/marble_part.cpp
--- marble-24.05.0/src/apps/marble-kde/marble_part.cpp.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-kde/marble_part.cpp	2024-05-28 20:28:59.229515740 +0200
@@ -132,7 +132,7 @@ MarblePart::MarblePart( QWidget *parentW
     // This has to happen before any initialization so plugins (for example) can
     // use it during initialization.
     MarbleLocale *marbleLocale = MarbleGlobal::getInstance()->locale();
-    // KF5 TODO: Read Connecting Calls to Catalogs" in ki18n programmer's guide
+    // KF6 TODO: Read Connecting Calls to Catalogs" in ki18n programmer's guide
     //           and enable translations
     //     KLocale *kLocale = KGlobal::locale();
 //     if ( kLocale->measureSystem() == KLocale::Metric ) {
diff -up marble-24.05.0/src/apps/marble-maps/CMakeLists.txt.6~ marble-24.05.0/src/apps/marble-maps/CMakeLists.txt
--- marble-24.05.0/src/apps/marble-maps/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-maps/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -6,7 +6,7 @@ include_directories(${CMAKE_CURRENT_BINA
 add_executable(marble-maps ${marble_SRCS} ${marble_QRCS})
 
 if (ANDROID)
-    find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED
+    find_package(Qt6 ${REQUIRED_QT_VERSION} REQUIRED
         COMPONENTS
             AndroidExtras
             QuickControls2
@@ -17,21 +17,21 @@ if (ANDROID)
         marble-maps
         marblewidget
         astro
-        Qt5::Concurrent
-        Qt5::Xml
-        Qt5::Widgets
-        Qt5::PrintSupport
-        Qt5::Network
-        Qt5::Sql
-        Qt5::Svg
-        Qt5::Positioning
-        Qt5::AndroidExtras
-        Qt5::Multimedia
-        Qt5::QuickControls2
+        Qt6::Concurrent
+        Qt6::Xml
+        Qt6::Widgets
+        Qt6::PrintSupport
+        Qt6::Network
+        Qt6::Sql
+        Qt6::Svg
+        Qt6::Positioning
+        Qt6::AndroidExtras
+        Qt6::Multimedia
+        Qt6::QuickControls2
     )
 
-    if (TARGET Qt5::WebEngineWidgets)
-        target_link_libraries(${TARGET} Qt5::WebEngineWidgets Qt5::WebEngine)
+    if (TARGET Qt6::WebEngineWidgets)
+        target_link_libraries(${TARGET} Qt6::WebEngineWidgets Qt6::WebEngine)
     endif()
 endif()
 
diff -up marble-24.05.0/src/apps/marble-maps/create-apk.py.6~ marble-24.05.0/src/apps/marble-maps/create-apk.py
--- marble-24.05.0/src/apps/marble-maps/create-apk.py.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-maps/create-apk.py	2024-05-28 20:28:59.229515740 +0200
@@ -66,9 +66,9 @@ class Filter(object):
 
         return []
 
-qtDir = os.environ.get('Qt5_android')
+qtDir = os.environ.get('Qt6_android')
 if qtDir is None:
-    print ('Please setup the Qt5_android environment variable point to your Qt installation')
+    print ('Please setup the Qt6_android environment variable point to your Qt installation')
     sys.exit(1)
 
 parser = argparse.ArgumentParser(description='Create an Android application package (APK) for Marble Maps')
diff -up marble-24.05.0/src/apps/marble-qt/CMakeLists.txt.6~ marble-24.05.0/src/apps/marble-qt/CMakeLists.txt
--- marble-24.05.0/src/apps/marble-qt/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-qt/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -25,9 +25,9 @@ endif (WIN32)
 
 target_link_libraries (
 marble-qt
-Qt5::Widgets
-Qt5::PrintSupport
-Qt5::Network
+Qt6::Widgets
+Qt6::PrintSupport
+Qt6::Network
 marbleui
 marblewidget
 astro)
diff -up marble-24.05.0/src/apps/marble-ui/CMakeLists.txt.6~ marble-24.05.0/src/apps/marble-ui/CMakeLists.txt
--- marble-24.05.0/src/apps/marble-ui/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/apps/marble-ui/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -15,15 +15,15 @@ target_link_libraries (
     marbleui
     PUBLIC
         marblewidget
-        Qt5::Widgets
-        Qt5::PrintSupport
+        Qt6::Widgets
+        Qt6::PrintSupport
     PRIVATE
-        Qt5::Network
+        Qt6::Network
 )
 if (NOT MARBLE_NO_DBUS)
     target_link_libraries( marbleui
         PRIVATE
-            Qt5::DBus
+            Qt6::DBus
     )
 endif ()
 
diff -up marble-24.05.0/src/bindings/CMakeLists.txt.6~ marble-24.05.0/src/bindings/CMakeLists.txt
--- marble-24.05.0/src/bindings/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/bindings/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -12,9 +12,9 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS Wallet NewStuff Parts)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS Wallet NewStuff Parts)
 
-if(NOT KF5_FOUND)
+if(NOT KF6_FOUND)
   return()
 endif()
 
diff -up marble-24.05.0/src/lib/marble/CMakeLists.txt.6~ marble-24.05.0/src/lib/marble/CMakeLists.txt
--- marble-24.05.0/src/lib/marble/CMakeLists.txt.6~	2024-05-28 20:28:59.201515437 +0200
+++ marble-24.05.0/src/lib/marble/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -1,14 +1,14 @@
 PROJECT(marblewidget VERSION ${RELEASE_SERVICE_VERSION})
 set(MARBLE_CMAKECONFIGNAME Marble)
 
-macro_optional_find_package(Phonon4Qt5 QUIET)
-marble_set_package_properties( Phonon4Qt5 PROPERTIES
+macro_optional_find_package(Phonon4Qt6 QUIET)
+marble_set_package_properties( Phonon4Qt6 PROPERTIES
     DESCRIPTION "Qt-based audio library"
     URL "https://phonon.kde.org/"
     PURPOSE "Support for playback of soundcue elements"
     TYPE OPTIONAL
 )
-if(Phonon4Qt5_FOUND)
+if(Phonon4Qt6_FOUND)
     set(HAVE_PHONON TRUE)
 endif()
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config-phonon.h.cmake
@@ -395,31 +395,31 @@ target_link_libraries(marblewidget PRIVA
 # link with release version of Qt libs
 target_link_libraries(marblewidget
     PUBLIC
-        Qt5::Xml
-        Qt5::Gui
-        Qt5::Widgets
-        Qt5::Network
+        Qt6::Xml
+        Qt6::Gui
+        Qt6::Widgets
+        Qt6::Network
     PRIVATE
-        Qt5::Core
-        Qt5::Svg
-        Qt5::PrintSupport
-        Qt5::Concurrent
+        Qt6::Core
+        Qt6::Svg
+        Qt6::PrintSupport
+        Qt6::Concurrent
 )
 if (NOT MARBLE_NO_WEBKITWIDGETS)
     target_link_libraries(marblewidget
         PUBLIC
-            Qt5::WebChannel
-            Qt5::WebEngineWidgets
-            Qt5::WebEngine
+            Qt6::WebChannel
+            Qt6::WebEngineWidgets
+            Qt6::WebEngine
     )
 endif ()
 if (NOT MARBLE_NO_DBUS)
     target_link_libraries(marblewidget
         PRIVATE
-            Qt5::DBus
+            Qt6::DBus
     )
 endif ()
-if (Phonon4Qt5_FOUND)
+if (Phonon4Qt6_FOUND)
     target_link_libraries(marblewidget
         PRIVATE
             Phonon::phonon4qt5
diff -up marble-24.05.0/src/lib/marble/MarbleConfig.cmake.in.6~ marble-24.05.0/src/lib/marble/MarbleConfig.cmake.in
--- marble-24.05.0/src/lib/marble/MarbleConfig.cmake.in.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/lib/marble/MarbleConfig.cmake.in	2024-05-28 20:28:59.229515740 +0200
@@ -5,18 +5,18 @@ set(MARBLE_NO_WEBKITWIDGETS @MARBLE_NO_W
 include(CMakeFindDependencyMacro OPTIONAL RESULT_VARIABLE _CMakeFindDependencyMacro_FOUND)
 
 if (_CMakeFindDependencyMacro_FOUND)
-    find_dependency(Qt5Xml @REQUIRED_QT_VERSION@)
-    find_dependency(Qt5Widgets @REQUIRED_QT_VERSION@)
+    find_dependency(Qt6Xml @REQUIRED_QT_VERSION@)
+    find_dependency(Qt6Widgets @REQUIRED_QT_VERSION@)
     if (NOT MARBLE_NO_WEBKITWIDGETS)
-        find_dependency(Qt5WebEngine @REQUIRED_QT_VERSION@)
-        find_dependency(Qt5WebEngineWidgets @REQUIRED_QT_VERSION@)
+        find_dependency(Qt6WebEngine @REQUIRED_QT_VERSION@)
+        find_dependency(Qt6WebEngineWidgets @REQUIRED_QT_VERSION@)
     endif()
 else()
-    find_package(Qt5Xml REQUIRED)
-    find_package(Qt5Widgets REQUIRED)
+    find_package(Qt6Xml REQUIRED)
+    find_package(Qt6Widgets REQUIRED)
     if (NOT MARBLE_NO_WEBKITWIDGETS)
-        find_package(Qt5WebEngine REQUIRED)
-        find_package(Qt5WebEngineWidgets REQUIRED)
+        find_package(Qt6WebEngine REQUIRED)
+        find_package(Qt6WebEngineWidgets REQUIRED)
     endif()
 endif()
 
diff -up marble-24.05.0/src/lib/marble/MarbleDirs.h.6~ marble-24.05.0/src/lib/marble/MarbleDirs.h
--- marble-24.05.0/src/lib/marble/MarbleDirs.h.6~	2024-05-28 20:28:59.205515479 +0200
+++ marble-24.05.0/src/lib/marble/MarbleDirs.h	2024-05-28 20:28:59.229515740 +0200
@@ -20,7 +20,7 @@ namespace Marble
  * @short A class that manages data look-up for Marble.
  *
  * The class basically does for Marble what KStandardDirs did for KDE4.
- * Given that the MarbleWidget is a Qt5-only library and given that it
+ * Given that the MarbleWidget is a Qt6-only library and given that it
  * comes with its own model and data we need this class.
  *
  * The class needs to respect the requirements of the different plattforms,
diff -up marble-24.05.0/src/lib/marble/declarative/CMakeLists.txt.6~ marble-24.05.0/src/lib/marble/declarative/CMakeLists.txt
--- marble-24.05.0/src/lib/marble/declarative/CMakeLists.txt.6~	2024-05-28 20:28:59.201515437 +0200
+++ marble-24.05.0/src/lib/marble/declarative/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -48,12 +48,12 @@ marble_add_project_resources(qmldir)
 target_link_libraries( marbledeclarative
     PUBLIC
         marblewidget
-        Qt5::Quick
+        Qt6::Quick
 )
-if(Qt5Positioning_FOUND)
+if(Qt6Positioning_FOUND)
     target_link_libraries( marbledeclarative
         PRIVATE
-            Qt5::Positioning
+            Qt6::Positioning
     )
 endif()
 
diff -up marble-24.05.0/src/lib/marble/kdescendantsproxymodel.h.6~ marble-24.05.0/src/lib/marble/kdescendantsproxymodel.h
--- marble-24.05.0/src/lib/marble/kdescendantsproxymodel.h.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/lib/marble/kdescendantsproxymodel.h	2024-05-28 20:28:59.229515740 +0200
@@ -11,7 +11,7 @@
 
 #include "marble_export.h"
 
-// namespace added to avoid symbol clashes with KF5::ItemModels
+// namespace added to avoid symbol clashes with KF6::ItemModels
 namespace Marble
 {
 
diff -up marble-24.05.0/src/plasma/CMakeLists.txt.6~ marble-24.05.0/src/plasma/CMakeLists.txt
--- marble-24.05.0/src/plasma/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plasma/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -13,9 +13,9 @@ set(SERVICES_INSTALL_DIR ${KDE_INSTALL_K
 set(DATA_INSTALL_DIR ${KDE_INSTALL_DATADIR})
 include(KDECMakeSettings)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS Plasma)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS Plasma)
 
-if (NOT KF5_FOUND)
+if (NOT KF6_FOUND)
   return()
 endif()
 
diff -up marble-24.05.0/src/plasmarunner/CMakeLists.txt.6~ marble-24.05.0/src/plasmarunner/CMakeLists.txt
--- marble-24.05.0/src/plasmarunner/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plasmarunner/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -12,9 +12,9 @@ include(KDEInstallDirs)
 set(LIB_INSTALL_DIR ${KDE_INSTALL_LIBDIR})
 include(KDECMakeSettings)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS Runner I18n)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS Runner I18n)
 
-if(NOT KF5_FOUND)
+if(NOT KF6_FOUND)
   return()
 endif()
 
@@ -26,12 +26,12 @@ set(plasmaRunner_SRCS
 add_library(plasma_runner_marble MODULE ${plasmaRunner_SRCS})
 target_link_libraries(plasma_runner_marble
   marblewidget
-  KF5::Runner
-  KF5::I18n
+  KF6::Runner
+  KF6::I18n
 )
 
 install(TARGETS plasma_runner_marble DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/krunner)
-if(KF5Runner_VERSION VERSION_LESS 5.72.0)
+if(KF6Runner_VERSION VERSION_LESS 5.72.0)
   install(FILES plasma-runner-marble.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
 else()
   kcoreaddons_desktop_to_json(plasma_runner_marble plasma-runner-marble.desktop)
diff -up marble-24.05.0/src/plugins/designer/CMakeLists.txt.6~ marble-24.05.0/src/plugins/designer/CMakeLists.txt
--- marble-24.05.0/src/plugins/designer/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/designer/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -1,21 +1,21 @@
 if (WITH_DESIGNER_PLUGIN)
 
-  # TODO: find why this is not listed in the feature_summary (same with Qt5UiPlugin)
-  find_package(Qt5 ${REQUIRED_QT_VERSION} QUIET COMPONENTS Designer)
-  marble_set_package_properties( Qt5Designer PROPERTIES TYPE OPTIONAL PURPOSE "Required to build Qt Designer plugins" )
+  # TODO: find why this is not listed in the feature_summary (same with Qt6UiPlugin)
+  find_package(Qt6 ${REQUIRED_QT_VERSION} QUIET COMPONENTS Designer)
+  marble_set_package_properties( Qt6Designer PROPERTIES TYPE OPTIONAL PURPOSE "Required to build Qt Designer plugins" )
 
-if(NOT Qt5Designer_VERSION VERSION_LESS 5.5.0)
-    find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG QUIET COMPONENTS UiPlugin)
-    set_package_properties(Qt5UiPlugin PROPERTIES TYPE OPTIONAL
+if(NOT Qt6Designer_VERSION VERSION_LESS 5.5.0)
+    find_package(Qt6 ${REQUIRED_QT_VERSION} CONFIG QUIET COMPONENTS UiPlugin)
+    set_package_properties(Qt6UiPlugin PROPERTIES TYPE OPTIONAL
         PURPOSE "Required to build Qt Designer plugins"
     )
-    if (Qt5UiPlugin_FOUND)
-        # for some reason, Qt5UiPlugin does not set its _INCLUDE_DIRS variable. Fill it manually
-        get_target_property(Qt5UiPlugin_INCLUDE_DIRS Qt5::UiPlugin INTERFACE_INCLUDE_DIRECTORIES)
+    if (Qt6UiPlugin_FOUND)
+        # for some reason, Qt6UiPlugin does not set its _INCLUDE_DIRS variable. Fill it manually
+        get_target_property(Qt6UiPlugin_INCLUDE_DIRS Qt6::UiPlugin INTERFACE_INCLUDE_DIRECTORIES)
     endif()
-    set(DesignerPluginDeps_FOUND ${Qt5UiPlugin_FOUND})
+    set(DesignerPluginDeps_FOUND ${Qt6UiPlugin_FOUND})
 else()
-    set(DesignerPluginDeps_FOUND ${Qt5Designer_FOUND})
+    set(DesignerPluginDeps_FOUND ${Qt6Designer_FOUND})
 endif()
 
   if (DesignerPluginDeps_FOUND)
diff -up marble-24.05.0/src/plugins/designer/latloneditwidget/CMakeLists.txt.6~ marble-24.05.0/src/plugins/designer/latloneditwidget/CMakeLists.txt
--- marble-24.05.0/src/plugins/designer/latloneditwidget/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/designer/latloneditwidget/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -3,8 +3,8 @@ PROJECT( LatLonEditPlugin )
 INCLUDE_DIRECTORIES(
  ${CMAKE_CURRENT_SOURCE_DIR}
  ${CMAKE_CURRENT_BINARY_DIR}
- ${Qt5UiPlugin_INCLUDE_DIRS}
- ${Qt5Designer_INCLUDE_DIRS}
+ ${Qt6UiPlugin_INCLUDE_DIRS}
+ ${Qt6Designer_INCLUDE_DIRS}
 )
 
 set( latloneditplugin_SRCS LatLonEditPlugin.cpp )
diff -up marble-24.05.0/src/plugins/designer/marblewidget/CMakeLists.txt.6~ marble-24.05.0/src/plugins/designer/marblewidget/CMakeLists.txt
--- marble-24.05.0/src/plugins/designer/marblewidget/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/designer/marblewidget/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -3,8 +3,8 @@ PROJECT( MarbleWidgetPlugin )
 INCLUDE_DIRECTORIES(
  ${CMAKE_CURRENT_SOURCE_DIR}
  ${CMAKE_CURRENT_BINARY_DIR}
- ${Qt5UiPlugin_INCLUDE_DIRS}
- ${Qt5Designer_INCLUDE_DIRS}
+ ${Qt6UiPlugin_INCLUDE_DIRS}
+ ${Qt6Designer_INCLUDE_DIRS}
 )
 
 set( marbleplugin_SRCS MarbleWidgetPlugin.cpp )
diff -up marble-24.05.0/src/plugins/designer/navigator/CMakeLists.txt.6~ marble-24.05.0/src/plugins/designer/navigator/CMakeLists.txt
--- marble-24.05.0/src/plugins/designer/navigator/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/designer/navigator/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -3,8 +3,8 @@ PROJECT(MarbleNavigatorPlugin)
 INCLUDE_DIRECTORIES(
  ${CMAKE_CURRENT_SOURCE_DIR}
  ${CMAKE_CURRENT_BINARY_DIR}
- ${Qt5UiPlugin_INCLUDE_DIRS}
- ${Qt5Designer_INCLUDE_DIRS}
+ ${Qt6UiPlugin_INCLUDE_DIRS}
+ ${Qt6Designer_INCLUDE_DIRS}
 )
 
 set( marblenavigator_SRCS MarbleNavigatorPlugin.cpp )
diff -up marble-24.05.0/src/plugins/positionprovider/CMakeLists.txt.6~ marble-24.05.0/src/plugins/positionprovider/CMakeLists.txt
--- marble-24.05.0/src/plugins/positionprovider/CMakeLists.txt.6~	2024-05-28 20:28:59.201515437 +0200
+++ marble-24.05.0/src/plugins/positionprovider/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -12,9 +12,9 @@ find_package(Qt6 ${REQUIRED_QT_VERSION}
     OPTIONAL_COMPONENTS
         Positioning
 )
-marble_set_package_properties( Qt5Positioning PROPERTIES DESCRIPTION "a collection of APIs and frameworks" )
-marble_set_package_properties( Qt5Positioning PROPERTIES URL "https://www.qt.io/developers/" )
-marble_set_package_properties( Qt5Positioning PROPERTIES TYPE OPTIONAL PURPOSE "position information via Qt5Positioning" )
+marble_set_package_properties( Qt6Positioning PROPERTIES DESCRIPTION "a collection of APIs and frameworks" )
+marble_set_package_properties( Qt6Positioning PROPERTIES URL "https://www.qt.io/developers/" )
+marble_set_package_properties( Qt6Positioning PROPERTIES TYPE OPTIONAL PURPOSE "position information via Qt6Positioning" )
 if(Qt6Positioning_FOUND)
    ADD_SUBDIRECTORY( qtpositioning )
 endif()
diff -up marble-24.05.0/src/plugins/positionprovider/qtpositioning/CMakeLists.txt.6~ marble-24.05.0/src/plugins/positionprovider/qtpositioning/CMakeLists.txt
--- marble-24.05.0/src/plugins/positionprovider/qtpositioning/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/positionprovider/qtpositioning/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,6 +9,6 @@ SET( qtpositioning_SRCS
  QtPositioningPositionProviderPlugin.cpp
 )
 
-SET( QtPositioningPositionProviderPlugin_LIBS Qt5::Positioning )
+SET( QtPositioningPositionProviderPlugin_LIBS Qt6::Positioning )
 
 MARBLE_ADD_PLUGIN( QtPositioningPositionProviderPlugin ${qtpositioning_SRCS} )
diff -up marble-24.05.0/src/plugins/positionprovider/wlocate/CMakeLists.txt.6~ marble-24.05.0/src/plugins/positionprovider/wlocate/CMakeLists.txt
--- marble-24.05.0/src/plugins/positionprovider/wlocate/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/positionprovider/wlocate/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -6,7 +6,7 @@ INCLUDE_DIRECTORIES(
  ${LIBWLOCATE_INCLUDE_DIR}
 )
 
-INCLUDE_DIRECTORIES(${Qt5Concurrent_INCLUDE_DIRS})
+INCLUDE_DIRECTORIES(${Qt6Concurrent_INCLUDE_DIRS})
 
 set( wlocate_SRCS
  WlocatePositionProviderPlugin.cpp
diff -up marble-24.05.0/src/plugins/render/aprs/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/aprs/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/aprs/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/aprs/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -1,18 +1,18 @@
 PROJECT(AprsPlugin)
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} COMPONENTS SerialPort QUIET)
-marble_set_package_properties( Qt5SerialPort PROPERTIES DESCRIPTION "Provides access to hardware and virtual serial ports." )
-marble_set_package_properties( Qt5SerialPort PROPERTIES TYPE OPTIONAL PURPOSE "Reading from serial port in APRS plugin" )
+find_package(Qt6 ${REQUIRED_QT_VERSION} COMPONENTS SerialPort QUIET)
+marble_set_package_properties( Qt6SerialPort PROPERTIES DESCRIPTION "Provides access to hardware and virtual serial ports." )
+marble_set_package_properties( Qt6SerialPort PROPERTIES TYPE OPTIONAL PURPOSE "Reading from serial port in APRS plugin" )
 
 INCLUDE_DIRECTORIES(
  ${CMAKE_CURRENT_SOURCE_DIR}
  ${CMAKE_CURRENT_BINARY_DIR}
 )
 
-if (Qt5SerialPort_FOUND)
-  include_directories (${Qt5SerialPort_INCLUDE_DIR})
+if (Qt6SerialPort_FOUND)
+  include_directories (${Qt6SerialPort_INCLUDE_DIR})
   set (serial_src AprsTTY.cpp)
-  list (APPEND AprsPlugin_LIBS ${Qt5SerialPort_LIBRARIES})
+  list (APPEND AprsPlugin_LIBS ${Qt6SerialPort_LIBRARIES})
   set (HAVE_QTSERIALPORT TRUE)
 endif ()
 
diff -up marble-24.05.0/src/plugins/render/compass/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/compass/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/compass/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/compass/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -10,6 +10,6 @@ set( compass_UI CompassConfigWidget.ui )
 qt_wrap_ui( compass_SRCS ${compass_UI} )
 qt_add_resources( compass_SRCS compass.qrc )
 
-set( CompassFloatItem_LIBS Qt5::Svg )
+set( CompassFloatItem_LIBS Qt6::Svg )
 
 marble_add_plugin( CompassFloatItem ${compass_SRCS} )
diff -up marble-24.05.0/src/plugins/render/crosshairs/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/crosshairs/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/crosshairs/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/crosshairs/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -10,6 +10,6 @@ set( crosshairs_UI CrosshairsConfigWidge
 qt_wrap_ui( crosshairs_SRCS ${crosshairs_UI} )
 qt_add_resources( crosshairs_SRCS crosshairs.qrc )
 
-set( CrosshairsPlugin_LIBS Qt5::Svg )
+set( CrosshairsPlugin_LIBS Qt6::Svg )
 
 marble_add_plugin( CrosshairsPlugin ${crosshairs_SRCS} )
diff -up marble-24.05.0/src/plugins/render/earthquake/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/earthquake/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/earthquake/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/earthquake/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -15,6 +15,6 @@ set( earthquake_UI EarthquakeConfigWidge
 qt_wrap_ui(my_SRCS ${earthquake_UI})
 qt_add_resources(my_SRCS earthquake.qrc)
 
-set(EarthquakePlugin_LIBS Qt5::Svg)
+set(EarthquakePlugin_LIBS Qt6::Svg)
 
 marble_add_plugin( EarthquakePlugin ${my_SRCS} )
diff -up marble-24.05.0/src/plugins/render/graticule/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/graticule/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/graticule/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/graticule/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,6 +9,6 @@ set( graticule_SRCS GraticulePlugin.cpp
 set( graticule_UI GraticuleConfigWidget.ui )
 qt_wrap_ui( graticule_SRCS ${graticule_UI} )
 
-set( GraticulePlugin_LIBS Qt5::Svg )
+set( GraticulePlugin_LIBS Qt6::Svg )
 
 marble_add_plugin( GraticulePlugin ${graticule_SRCS} )
diff -up marble-24.05.0/src/plugins/render/overviewmap/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/overviewmap/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/overviewmap/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/overviewmap/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -10,6 +10,6 @@ set( overviewmap_UI OverviewMapConfigWid
 
 qt_wrap_ui(overviewmap_SRCS ${overviewmap_UI})
 
-set( OverviewMap_LIBS Qt5::Svg )
+set( OverviewMap_LIBS Qt6::Svg )
 
 marble_add_plugin( OverviewMap ${overviewmap_SRCS} )
diff -up marble-24.05.0/src/plugins/render/routing/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/routing/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/routing/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/routing/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -1,7 +1,7 @@
 PROJECT(RoutingPlugin)
 
-macro_optional_find_package(Phonon4Qt5 QUIET)
-marble_set_package_properties( Phonon4Qt5 PROPERTIES
+macro_optional_find_package(Phonon4Qt6 QUIET)
+marble_set_package_properties( Phonon4Qt6 PROPERTIES
     DESCRIPTION "Qt-based audio library"
     URL "https://phonon.kde.org/"
     PURPOSE "Voice navigation (sound or human speakers)"
@@ -13,7 +13,7 @@ INCLUDE_DIRECTORIES(
  ${CMAKE_CURRENT_BINARY_DIR}
 )
 
-if(Phonon4Qt5_FOUND)
+if(Phonon4Qt6_FOUND)
   set( routing_SRCS RoutingPlugin.cpp AudioOutput.cpp )
 ELSE()
   set( routing_SRCS RoutingPlugin.cpp NullAudioOutput.cpp )
@@ -23,6 +23,6 @@ qt_wrap_ui( routing_SRCS ${routing_UI} )
 qt_add_resources(routing_SRCS routing.qrc)
 
 marble_add_plugin( RoutingPlugin ${routing_SRCS} )
-if(Phonon4Qt5_FOUND)
+if(Phonon4Qt6_FOUND)
     target_link_libraries(RoutingPlugin Phonon::phonon4qt5)
 endif()
diff -up marble-24.05.0/src/plugins/render/weather/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/weather/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/weather/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/weather/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -25,6 +25,6 @@ set( weather_UI WeatherConfigWidget.ui )
 qt_wrap_ui( weather_SRCS ${weather_UI} )
 qt_add_resources( weather_SRCS weather.qrc )
 
-set(Weather_LIBS Qt5::Svg)
+set(Weather_LIBS Qt6::Svg)
 
 marble_add_plugin( Weather ${weather_SRCS} )
diff -up marble-24.05.0/src/plugins/render/wikipedia/CMakeLists.txt.6~ marble-24.05.0/src/plugins/render/wikipedia/CMakeLists.txt
--- marble-24.05.0/src/plugins/render/wikipedia/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/render/wikipedia/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -13,6 +13,6 @@ set( wikipedia_UI WikipediaConfigWidget.
 
 qt_wrap_ui(wikipedia_SRCS ${wikipedia_UI})
 
-set( Wikipedia_LIBS Qt5::Svg )
+set( Wikipedia_LIBS Qt6::Svg )
 
 marble_add_plugin( Wikipedia ${wikipedia_SRCS} )
diff -up marble-24.05.0/src/plugins/runner/gpx/CMakeLists.txt.6~ marble-24.05.0/src/plugins/runner/gpx/CMakeLists.txt
--- marble-24.05.0/src/plugins/runner/gpx/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/runner/gpx/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -40,7 +40,7 @@ if( BUILD_MARBLE_TESTS )
     set( TestTrack_SRCS TestTrack.moc ${TestTrack_SRCS} )
 
     add_executable( TestTrack ${TestTrack_SRCS} )
-    target_link_libraries( TestTrack Qt5::Test
+    target_link_libraries( TestTrack Qt6::Test
                                      marblewidget )
     set_target_properties( TestTrack PROPERTIES
                             COMPILE_FLAGS "-DDATA_PATH=\"\\\"${DATA_PATH}\\\"\" -DPLUGIN_PATH=\"\\\"${PLUGIN_PATH}\\\"\"" )
@@ -57,8 +57,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS KIO)
-if(NOT KF5_FOUND)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS KIO)
+if(NOT KF6_FOUND)
     return()
 endif()
 
diff -up marble-24.05.0/src/plugins/runner/json/CMakeLists.txt.6~ marble-24.05.0/src/plugins/runner/json/CMakeLists.txt
--- marble-24.05.0/src/plugins/runner/json/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/runner/json/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -18,8 +18,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS KIO)
-if(NOT KF5_FOUND)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS KIO)
+if(NOT KF6_FOUND)
     return()
 endif()
 
diff -up marble-24.05.0/src/plugins/runner/kml/CMakeLists.txt.6~ marble-24.05.0/src/plugins/runner/kml/CMakeLists.txt
--- marble-24.05.0/src/plugins/runner/kml/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/runner/kml/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -20,22 +20,22 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS KIO)
-if(NOT KF5_FOUND)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS KIO)
+if(NOT KF6_FOUND)
     return()
 endif()
 
 # register marble part as handler
 # install again once part is fixed to load and show the passed file
-# install(FILES marble_part_kml.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
-# install(FILES marble_part_kmz.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
+# install(FILES marble_part_kml.desktop DESTINATION ${KDE_INSTALL_FULL_KSERVICESDIR})
+# install(FILES marble_part_kmz.desktop DESTINATION ${KDE_INSTALL_FULL_KSERVICESDIR})
 
 # register thumbnail plugin as handler
-install(FILES marble_thumbnail_kml.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
-install(FILES marble_thumbnail_kmz.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
+install(FILES marble_thumbnail_kml.desktop DESTINATION ${KDE_INSTALL_FULL_KSERVICESDIR})
+install(FILES marble_thumbnail_kmz.desktop DESTINATION ${KDE_INSTALL_FULL_KSERVICESDIR})
 
 if(NOT WIN32 AND NOT APPLE)
     # register marble app as handler
-    install(FILES marble_kml.desktop DESTINATION ${KDE_INSTALL_APPDIR})
-    install(FILES marble_kmz.desktop DESTINATION ${KDE_INSTALL_APPDIR})
+    install(FILES marble_kml.desktop DESTINATION ${KDE_INSTALL_FULL_APPDIR})
+    install(FILES marble_kmz.desktop DESTINATION ${KDE_INSTALL_FULL_APPDIR})
 endif()
diff -up marble-24.05.0/src/plugins/runner/local-osm-search/CMakeLists.txt.6~ marble-24.05.0/src/plugins/runner/local-osm-search/CMakeLists.txt
--- marble-24.05.0/src/plugins/runner/local-osm-search/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/runner/local-osm-search/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -14,4 +14,4 @@ DatabaseQuery.cpp
  )
 
 marble_add_plugin( LocalOsmSearchPlugin ${localOsmSearch_SRCS} )
-target_link_libraries( LocalOsmSearchPlugin Qt5::Sql )
+target_link_libraries( LocalOsmSearchPlugin Qt6::Sql )
diff -up marble-24.05.0/src/plugins/runner/osm/CMakeLists.txt.6~ marble-24.05.0/src/plugins/runner/osm/CMakeLists.txt
--- marble-24.05.0/src/plugins/runner/osm/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/runner/osm/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -61,8 +61,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS KIO)
-if(NOT KF5_FOUND)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS KIO)
+if(NOT KF6_FOUND)
     return()
 endif()
 
diff -up marble-24.05.0/src/plugins/runner/shp/CMakeLists.txt.6~ marble-24.05.0/src/plugins/runner/shp/CMakeLists.txt
--- marble-24.05.0/src/plugins/runner/shp/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/plugins/runner/shp/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -22,8 +22,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS KIO)
-if(NOT KF5_FOUND)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS KIO)
+if(NOT KF6_FOUND)
     return()
 endif()
 
diff -up marble-24.05.0/src/thumbnailer/CMakeLists.txt.6~ marble-24.05.0/src/thumbnailer/CMakeLists.txt
--- marble-24.05.0/src/thumbnailer/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/src/thumbnailer/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -8,9 +8,9 @@ endif()
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
 include(KDEInstallDirs)
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS KIO)
+macro_optional_find_package(KF6 ${REQUIRED_KF6_MIN_VERSION} QUIET COMPONENTS KIO)
 
-if (NOT KF5_FOUND)
+if (NOT KF6_FOUND)
   return()
 endif()
 
@@ -21,6 +21,6 @@ set(marblethumbnail_SRCS
 add_library(marblethumbnail MODULE ${marblethumbnail_SRCS})
 target_link_libraries(marblethumbnail
   marblewidget
-  KF5::KIOWidgets
+  KF6::KIOWidgets
 )
 install(TARGETS marblethumbnail  DESTINATION ${KDE_INSTALL_PLUGINDIR})
diff -up marble-24.05.0/tests/CTestNightlyScript.cmake.in.6~ marble-24.05.0/tests/CTestNightlyScript.cmake.in
--- marble-24.05.0/tests/CTestNightlyScript.cmake.in.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tests/CTestNightlyScript.cmake.in	2024-05-28 20:28:59.229515740 +0200
@@ -6,7 +6,7 @@ set( CTEST_BINARY_DIRECTORY "@PROJECT_BI
 set ( CTEST_COMMAND "@CMAKE_CTEST_COMMAND@ -D Nightly --verbose" )
 
 #set the command for configuring the dashboard
-set ( CTEST_CMAKE_COMMAND "@CMAKE_COMMAND@ -DWITH_KF5=FALSE" )
+set ( CTEST_CMAKE_COMMAND "@CMAKE_COMMAND@ -DWITH_KF6=FALSE" )
 
 #wipe the binary tree before running
 set ( CTEST_START_WITH_EMPTY_BINARY_DIRECTORY TRUE )
diff -up marble-24.05.0/tools/asc2kml/CMakeLists.txt.6~ marble-24.05.0/tools/asc2kml/CMakeLists.txt
--- marble-24.05.0/tools/asc2kml/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/asc2kml/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC asc2kml.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core )
+target_link_libraries( ${TARGET} Qt6::Core )
diff -up marble-24.05.0/tools/dso2kml/CMakeLists.txt.6~ marble-24.05.0/tools/dso2kml/CMakeLists.txt
--- marble-24.05.0/tools/dso2kml/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/dso2kml/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC dso2kml.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core )
+target_link_libraries( ${TARGET} Qt6::Core )
diff -up marble-24.05.0/tools/iau2kml/CMakeLists.txt.6~ marble-24.05.0/tools/iau2kml/CMakeLists.txt
--- marble-24.05.0/tools/iau2kml/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/iau2kml/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC iau2kml.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core )
+target_link_libraries( ${TARGET} Qt6::Core )
diff -up marble-24.05.0/tools/mbtile-import/CMakeLists.txt.6~ marble-24.05.0/tools/mbtile-import/CMakeLists.txt
--- marble-24.05.0/tools/mbtile-import/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/mbtile-import/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC mbtile-import.cpp MbTileWriter.cpp)
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries(${TARGET} marblewidget Qt5::Sql)
+target_link_libraries(${TARGET} marblewidget Qt6::Sql)
diff -up marble-24.05.0/tools/osm-addresses/CMakeLists.txt.6~ marble-24.05.0/tools/osm-addresses/CMakeLists.txt
--- marble-24.05.0/tools/osm-addresses/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/osm-addresses/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -29,7 +29,7 @@ add_executable( ${TARGET} ${${TARGET}_SR
 
 target_link_libraries( ${TARGET}
     marblewidget
-    Qt5::Sql
+    Qt6::Sql
     ${PROTOBUF_LIBRARIES}
     ${ZLIB_LIBRARIES}
 )
diff -up marble-24.05.0/tools/osm-sisyphus/CMakeLists.txt.6~ marble-24.05.0/tools/osm-sisyphus/CMakeLists.txt
--- marble-24.05.0/tools/osm-sisyphus/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/osm-sisyphus/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -18,4 +18,4 @@ main.cpp
 )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries(${TARGET} marblewidget Qt5::Sql)
+target_link_libraries(${TARGET} marblewidget Qt6::Sql)
diff -up marble-24.05.0/tools/pnt2svg/CMakeLists.txt.6~ marble-24.05.0/tools/pnt2svg/CMakeLists.txt
--- marble-24.05.0/tools/pnt2svg/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/pnt2svg/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC pnt2svg.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core )
+target_link_libraries( ${TARGET} Qt6::Core )
diff -up marble-24.05.0/tools/pntdel/CMakeLists.txt.6~ marble-24.05.0/tools/pntdel/CMakeLists.txt
--- marble-24.05.0/tools/pntdel/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/pntdel/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC pntdel.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core )
+target_link_libraries( ${TARGET} Qt6::Core )
diff -up marble-24.05.0/tools/pntreplace/CMakeLists.txt.6~ marble-24.05.0/tools/pntreplace/CMakeLists.txt
--- marble-24.05.0/tools/pntreplace/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/pntreplace/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC pntreplace.cpp svgxmlhandler.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Xml Qt5::Widgets )
+target_link_libraries( ${TARGET} Qt6::Xml Qt6::Widgets )
diff -up marble-24.05.0/tools/routing-instructions/CMakeLists.txt.6~ marble-24.05.0/tools/routing-instructions/CMakeLists.txt
--- marble-24.05.0/tools/routing-instructions/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/routing-instructions/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -15,4 +15,4 @@ set( routing-instructions_SRC
 
 add_executable( routing-instructions ${routing-instructions_SRC} )
 
-target_link_libraries( routing-instructions Qt5::Core )
+target_link_libraries( routing-instructions Qt6::Core )
diff -up marble-24.05.0/tools/sentineltile/CMakeLists.txt.6~ marble-24.05.0/tools/sentineltile/CMakeLists.txt
--- marble-24.05.0/tools/sentineltile/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/sentineltile/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC sentineltile.cpp tileprocessor.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET}  Qt5::Core Qt5::Widgets )
+target_link_libraries( ${TARGET}  Qt6::Core Qt6::Widgets )
diff -up marble-24.05.0/tools/speaker-files/CMakeLists.txt.6~ marble-24.05.0/tools/speaker-files/CMakeLists.txt
--- marble-24.05.0/tools/speaker-files/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/speaker-files/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC main.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core Qt5::Sql )
+target_link_libraries( ${TARGET} Qt6::Core Qt6::Sql )
diff -up marble-24.05.0/tools/stars/CMakeLists.txt.6~ marble-24.05.0/tools/stars/CMakeLists.txt
--- marble-24.05.0/tools/stars/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/stars/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -10,4 +10,4 @@ set( ${TARGET}_SRC stars.cpp )
 add_definitions( -D_USE_MATH_DEFINES=1 )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET} Qt5::Core )
+target_link_libraries( ${TARGET} Qt6::Core )
diff -up marble-24.05.0/tools/svg2pnt/CMakeLists.txt.6~ marble-24.05.0/tools/svg2pnt/CMakeLists.txt
--- marble-24.05.0/tools/svg2pnt/CMakeLists.txt.6~	2024-05-17 00:07:47.000000000 +0200
+++ marble-24.05.0/tools/svg2pnt/CMakeLists.txt	2024-05-28 20:28:59.229515740 +0200
@@ -9,4 +9,4 @@ include_directories(
 set( ${TARGET}_SRC svg2pnt.cpp svgxmlhandler.cpp )
 add_executable( ${TARGET} ${${TARGET}_SRC} )
 
-target_link_libraries( ${TARGET}  Qt5::Core Qt5::Xml )
+target_link_libraries( ${TARGET}  Qt6::Core Qt6::Xml )
diff -up marble-24.05.0/tools/vectorosm-tilecreator/CMakeLists.txt.6~ marble-24.05.0/tools/vectorosm-tilecreator/CMakeLists.txt
--- marble-24.05.0/tools/vectorosm-tilecreator/CMakeLists.txt.6~	2024-05-28 20:28:59.229515740 +0200
+++ marble-24.05.0/tools/vectorosm-tilecreator/CMakeLists.txt	2024-05-28 20:33:51.868718703 +0200
@@ -25,7 +25,7 @@ TileDirectory.cpp
 VectorClipper.cpp
 )
 target_compile_definitions(vectorosm-toolchain PUBLIC USINGZ=1)
-target_link_libraries(vectorosm-toolchain marblewidget Qt5::Sql)
+target_link_libraries(vectorosm-toolchain marblewidget Qt6::Sql)
 
 add_executable(marble-vectorosm-tilecreator
     SpellChecker.cpp
