diff -up marble-24.12.0/src/lib/marble/declarative/CMakeLists.txt.1~ marble-24.12.0/src/lib/marble/declarative/CMakeLists.txt
--- marble-24.12.0/src/lib/marble/declarative/CMakeLists.txt.1~	2024-12-15 18:53:06.189062227 +0100
+++ marble-24.12.0/src/lib/marble/declarative/CMakeLists.txt	2024-12-15 18:55:30.520339348 +0100
@@ -74,6 +74,13 @@ target_sources(marbledeclarative PRIVATE
     RouteRelationModel.h
 )
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
+  SET_TARGET_PROPERTIES(marbledeclarative PROPERTIES
+    VERSION 0.0.1
+    SOVERSION 0
+  )
+endif()
+
 target_link_libraries(marbledeclarative PUBLIC
     marblewidget
     Qt6::Quick
