diff -Naur marble-17.04.1/src/lib/marble/declarative/CMakeLists.txt marble-17.04.1.tpg/src/lib/marble/declarative/CMakeLists.txt
--- marble-17.04.1/src/lib/marble/declarative/CMakeLists.txt	2017-05-07 13:57:55.000000000 +0000
+++ marble-17.04.1.tpg/src/lib/marble/declarative/CMakeLists.txt	2017-05-12 17:23:29.272944523 +0000
@@ -30,6 +30,13 @@
     RouteRelationModel.cpp
 )
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
+  SET_TARGET_PROPERTIES(marbledeclarative PROPERTIES
+    VERSION 0.0.1
+    SOVERSION 0
+  )
+endif()
+
 generate_export_header(marbledeclarative BASE_NAME marble_declarative)
 target_include_directories(marbledeclarative PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 
