diff -Naur marble-15.12.2/src/3rdparty/sgp4/sgp4ext.cpp marble-15.12.2.tpg/src/3rdparty/sgp4/sgp4ext.cpp
--- marble-15.12.2/src/3rdparty/sgp4/sgp4ext.cpp	2016-01-31 19:57:21.000000000 +0000
+++ marble-15.12.2.tpg/src/3rdparty/sgp4/sgp4ext.cpp	2016-03-03 18:18:20.049989107 +0000
@@ -207,7 +207,7 @@
 *
 * --------------------------------------------------------------------------- */
 
-double  asinh
+double  std::asinh
         (
           double xval
         )
