diff -urN marble-16.08.2/src/lib/marble/declarative/CMakeLists.txt marble-16.08.2-patched/src/lib/marble/declarative/CMakeLists.txt
--- marble-16.08.2/src/lib/marble/declarative/CMakeLists.txt	2016-09-26 10:37:20.000000000 +1000
+++ marble-16.08.2-patched/src/lib/marble/declarative/CMakeLists.txt	2016-10-28 20:36:10.430875782 +1000
@@ -32,6 +32,13 @@
     Settings.cpp
 )
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
+  SET_TARGET_PROPERTIES(marbledeclarative PROPERTIES
+    VERSION 0.0.1
+    SOVERSION 0
+  )
+endif()
+
 generate_export_header(marbledeclarative BASE_NAME marble_declarative)
 target_include_directories(marbledeclarative PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 
