diff -up marble-23.03.90/CMakeLists.txt.omv~ marble-23.03.90/CMakeLists.txt
--- marble-23.03.90/CMakeLists.txt.omv~	2023-04-14 23:12:21.478897681 +0200
+++ marble-23.03.90/CMakeLists.txt	2023-04-14 23:12:35.289009636 +0200
@@ -32,7 +32,7 @@ if ("${CMAKE_C_COMPILER_ID}" STREQUAL "G
     set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -std=iso9899:1990")
 endif()
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -fno-exceptions")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fno-exceptions")
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" AND NOT WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
 endif()
diff -up marble-23.03.90/src/plugins/runner/osm/OsmPbfParser.cpp.omv~ marble-23.03.90/src/plugins/runner/osm/OsmPbfParser.cpp
--- marble-23.03.90/src/plugins/runner/osm/OsmPbfParser.cpp.omv~	2023-04-14 23:11:37.146539115 +0200
+++ marble-23.03.90/src/plugins/runner/osm/OsmPbfParser.cpp	2023-04-14 23:11:52.290661456 +0200
@@ -7,6 +7,7 @@
 #include "OsmPbfParser.h"
 
 #ifdef HAVE_PROTOBUF
+#include <absl/strings/string_view.h>
 #include "fileformat.pb.h"
 #include "osmformat.pb.h"
 #endif
